src/database/schema/ChatSchema.ts

```diff
@@ -171,14 +177,21 @@ class ChatSchema implements BaseDbSchema {
   }
 
   static fromDatabaseObject(dbObject: any): ChatSchema {
-    let rawJson = null;
+    let rawJson: string | null = null;
+    let attachmentJson: string | null = null;
 
     try {
       rawJson = JSON.parse(dbObject?.raw_json || '{}');
     } catch (e) {
       console.log('error parse');
       console.log(e);
     }
+    try {
+      attachmentJson = JSON.parse(dbObject?.attachment_json || '[]');
+    } catch (e) {
+      console.log('error parse');
+      console.log(e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling for parsing `rawJson` and `attachmentJson` is done separately, which leads to code duplication. Consider creating a helper function to parse JSON and handle errors.
```diff
-    let rawJson: string | null = null;
-    let attachmentJson: string | null = null;
-  
-     try {
-       rawJson = JSON.parse(dbObject?.raw_json || '{}');
-     } catch (e) {
-       console.log('error parse');
-       console.log(e);
-     }
-    try {
-      attachmentJson = JSON.parse(dbObject?.attachment_json || '[]');
-    } catch (e) {
-      console.log('error parse');
-      console.log(e);
-    }
+    const parseJson = (jsonString: string, defaultValue: any) => {
+      try {
+        return JSON.parse(jsonString);
+      } catch (e) {
+        console.log('error parse', e);
+        return defaultValue;
+      }
+    };
+
+    let rawJson = parseJson(dbObject?.raw_json, {});
+    let attachmentJson = parseJson(dbObject?.attachment_json, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
No need, because only in this file